src/main/java/graphics/Window.java

```diff
@@ -180,8 +180,8 @@ public void showWindow() {
                 PostProcessing.prepare();
                 sceneManager.postProcess(currentScene().renderer.fetchColorAttachment(0));
                 PostProcessing.finish();
-                sceneManager.debugRender();
                 sceneManager.updateUI();
+                sceneManager.debugRender();

```

Comment:
do we rly still wanna do the debug and UI update after update+draw of everything else?

Reply:
yes, post processing on the scene overwrites the buffer so UI and debug need to be drawn after so they are not overwritten.
The UI however probably should be drawn over the debug, so the order there could be swapped.

Reply:
I see, well makes sense. About the UI I am not sure, what if you wanna see the debug over the UI?